src/cmd-buildextend-installer

```diff
@@ -149,17 +149,19 @@ def generate_iso():
     initramfs_img = 'initramfs.img'
 
     tmpisofile = os.path.join(tmpdir, iso_name)
-    img_metal_obj = buildmeta['images'].get('metal')
-    if img_metal_obj is None:
-        raise Exception("ISO generation requires `metal` image")
-
-    img_metal = os.path.join(builddir, img_metal_obj['path'])
-    img_metal_checksum = img_metal_obj['sha256']
 
+    img_metal_obj = buildmeta['images'].get('metal')
     img_metal4k_obj = buildmeta['images'].get('metal4k')
-    if img_metal4k_obj is not None:
+    if is_live:
+        if img_metal_obj is None or img_metal4k_obj is None:
+            raise Exception("Live image generation requires `metal` and `metal4k` images")
```

Comment:
Not a blocker, but I think we should have a switch like `--allow-missing-metal4k` to still allow skipping this for the local dev case. Iterating on the live ISO is super painful, and while the majority of the time is the `mksquashfs` step, having to recreate a metal4k image we don't care about each time accumulates to a non-trivial amount of time.

Reply:
```
<jlebon> i'd say we can try getting it in as is, and if someone (perhaps me) gets annoyed enough
         in the future to add that option, i wouldn't oppose :)
```

:+1: Let's skip for now.  If we want to re-add the option, I agree that `--allow-missing-...` is more palatable, since the caller knows they're asking for that, and it doesn't matter as much if it breaks.